statement_args.go

```diff
@@ -49,10 +49,27 @@ func quoteNeeded(a interface{}) bool {
 	return true
 }
 
-func convertArg(arg interface{}) string {
+func convertStringSingleQuote(arg string) string {
+	return "'" + strings.Replace(arg, "'", "''", -1) + "'"
+}
+
+func convertString(arg string) string {
+	var buf strings.Builder
+	buf.WriteRune('\'')
+	for _, c := range arg {
+		if c == '\\' || c == '\'' {
+			buf.WriteRune('\\')
+		}
+		buf.WriteRune(c)
+	}
+	buf.WriteRune('\'')
+	return buf.String()
+}
+
+func convertArg(arg interface{}, convertFunc func(string) string) string {
 	if quoteNeeded(arg) {
```

Comment:
What I don't understand is why traditional positional parameters like `(?, ?, ?)` don't work here.

Reply:
A good question, let me try.